server/src/main/java/org/opensearch/snapshots/SnapshotInfo.java

```diff
@@ -694,7 +695,7 @@ private XContentBuilder toXContentInternal(final XContentBuilder builder, final
         builder.field(UUID, snapshotId.getUUID());
         assert version != null : "version must always be known when writing a snapshot metadata blob";
         builder.field(VERSION_ID, version.id);
-        if (remoteStoreIndexShallowCopy != null) {
+        if (FeatureFlags.isEnabled(FeatureFlags.REMOTE_STORE) && remoteStoreIndexShallowCopy != null) {
```

Comment:
is this used for API or the serialization [here](https://github.com/opensearch-project/OpenSearch/pull/8594/files#diff-e6fa71852f7522df91474eb81a58eed36f7f410fbe8e33816c8cc1978d9c0928R641) is used for API  ? We want to just control the API changes whereas in the blob it will still be preserved right ?

Reply:
actually just removing this from API parameter is a bit tricky, currently adding this feature flag check at all places which should be fine now. later when we make this GA, we anyway have to remove these feature flag checks.